howtos/how-to-make-clickable-tabs-for-conversations/index.html

```diff
@@ -0,0 +1,150 @@
+<!DOCTYPE html>
+
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+
+    <title>TalkJS tutorial</title>
+
+    <link
+      rel="stylesheet"
+      href="https://fonts.googleapis.com/css2?family=Inter:wght@400;600;700"
+    />
+    <link
+      rel="stylesheet"
+      href="styles.css"
+    />
+  </head>
+
+  <body>
+    <div id="chat-container">
+      <div id="dynamicTabContainer" class="tab">
+        <!-- Tabs will be added here -->
+      </div>
+      <!-- container element in which TalkJS will display a chat UI -->
+      <div id="talkjs-container">
+        <i>Loading chat...</i>
+      </div>
+    </div>
+  </body>
+</html>
+
+<script>
+  (function (t, a, l, k, j, s) {
+    s = a.createElement("script");
+    s.async = 1;
+    s.src = "https://cdn.talkjs.com/talk.js";
+    a.head.appendChild(s);
+    k = t.Promise;
+    t.Talk = {
+      v: 3,
+      ready: {
+        then: function (f) {
+          if (k)
+            return new k(function (r, e) {
+              l.push([f, r, e]);
+            });
+          l.push([f]);
+        },
+        catch: function () {
+          return k && new k();
+        },
+        c: l,
+      },
+    };
+  })(window, document, []);
+</script>
+
+<script>
+  let chatbox;
+
+  Talk.ready.then(function () {
+    const me = new Talk.User({
+      id: "tabConversationExampleSupportAgent",
+      name: "Alice",
+      email: "alice@example.com",
+      role: "default",
+      photoUrl: "https://talkjs.com/images/avatar-1.jpg",
+      welcomeMessage: "Hey there! How can I help?",
+    });
+
+    const talkSession = new Talk.Session({
+      appId: "<APP_ID>", // replace with your app ID
+      me: me,
+    });
+
+    chatbox = talkSession.createChatbox();
+    chatbox.mount(document.getElementById("talkjs-container"));
+
+    chatbox.onCustomMessageAction("openTab", (event) => {
+    //The handler for our open tab custom action. This event
+    //creates the tab for the user and swaps to a conversation with the selected user.
+      const messageSender = new Talk.User({
+          id: event.message.senderId,
+          name: event.message.sender.name,
+          email: event.message.sender.email,
+          role: event.message.sender.role,
+          photoUrl: event.message.sender.photoUrl,
+          welcomeMessage: event.message.sender.welcomeMessage,
+      });
+      
+      const conversation = talkSession.getOrCreateConversation(
+          event.message.senderId
```

Comment:
When I click "Open tab" I'd intuitively expect it to open a 1-1 conversation between me and the sender. However, the conversation's ID here is being set to `event.message.senderId`.

This means that if _N_ people click "Open tab" on Bob's message, they're all added to one chat instead of there being _N_ independent 1-1 chats with Bob.

The conversation ID should be generated with [`Talk.oneOnOneId()`](https://talkjs.com/docs/Reference/JavaScript_Chat_SDK/The_Talk_Object/#Talk.oneOnOneId) like so:

```js
Talk.oneOnOneId(me, event.message.senderId)
```

Reply:
Done!